compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaObjectTypeSymbol.java

```diff
@@ -169,6 +170,24 @@ public String signature() {
                 .toString();
     }
 
+    /**
+     * As per the spec, {@code lang.value} functions should not be available on objects. Therefore, they are filtered
+     * here.
+     *
+     * @param functions    Functions to be filtered
+     * @param internalType Internal type
+     * @return Filtered langlib functions
+     */
+    @Override
+    protected List<FunctionSymbol> filterLangLibMethods(List<FunctionSymbol> functions, BType internalType) {
+        List<FunctionSymbol> functionSymbols = super.filterLangLibMethods(functions, internalType);
+        return functionSymbols.stream()
+                .filter(functionSymbol -> functionSymbol.getModule().isPresent())
+                .filter(functionSymbol -> !"ballerina".equals(functionSymbol.getModule().get().id().orgName()) ||
```

Comment:
If we have constants for these `ballerina` and `lang.value` let's use them

Reply:
Done!